### ISSUE TITLE
SharePoint Framework Web Part not working properly in Viva Connections

### ISSUE THREAD

**Issue by tripleseven73** [SharePoint/sp-dev-docs#7145]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [X] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

SPFx version 1.11.0

### Describe the bug / error

Using Viva Connections we have added one of our SharePoint Online sites to the Teams App Bar. If we navigate to a SharePoint Modern Page using de App Bar Menu, our SharePoint Framework Web Parts on that page do not work properly.

Among things we've noticed thus far:
- onActiveItemChanged event of DetailList Component not firing
- SharePoint Theme is not being applied

If we use the App Bar to navigate to a different page and from there use SharePoints own menu's to navigate to the page containing the Web Parts, everything works as expected. It somehow feels similar to the partial page loading issue involving the SharePoint left vertical menu.

### Steps to reproduce

Steps to reproduce:
- using Viva Connections, create an App in de Teams App Bar pointing to a Modern Team or Hub Site
- have a Modern Page with a SharePoint Web Part displaying a DetailList and a Button
- make sure this Modern Page is accessible in e menu (e.g. horizontal navigation)
- open the SharePoint site using the App in de Teams App Bar
- click on the App for a second time to open the App Menu
- open the menu item pointing to the Modern Page containing the Web Part
- verify that the DetailList onActiveItemChanged event will not fire

We've have been able to reproduce this on all our SharePoint Online tenants using a very basic SPFx Web Part created with SPFx v1.11.0.

### Expected behavior

Web Part should function the same when using SharePoint in Teams using Viva Connections.

**Comment by patmill** [comment]
Hi @tripleseven73 - just to completely confirm things here.  Is the problem that extensions that you have installed on a list aren't working on the list webpart when it is hosted in the Viva Connections app?  Or is it that you have a custom webpart using fabric react details list and it isn't firing properly?

Can you share out a simple project that shows the issue?

**Comment by tripleseven73** [comment]
Hi @patmill ,
It is the latter: a custom Web Part using a fabric react details list. I have attached the SPFx solution containing the ShowList Web Part I'm using to reproduce the issue.

Important note: In the original bug report I mentioned the use of SPFx v1.11.0. I recently upgraded all our SPFx solutions to v1.12.1.  This includes the ShowList Web Part I have attached. The issue is still present after upgrade.

[csgDetailListInVivaConnections.zip](https://github.com/SharePoint/sp-dev-docs/files/6984297/csgDetailListInVivaConnections.zip)
